src/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,28 @@
+import { useEffect, useState } from "react"
+
+// This custom hook allows to read data from localStorage to initialize state
+export const useLocalStorage = (key, initialValue) => {
+  const [storedValue, setStoredValue] = useState(initialValue)
+  const setValue = (value) => {
+    try {
+      const valueToStore =
+        value instanceof Function ? value(storedValue) : value
+      setStoredValue(valueToStore)
+      window.localStorage.setItem(key, JSON.stringify(valueToStore))
+    } catch (error) {
+      console.log(error)
+    }
+  }
+
+  useEffect(() => {
+    try {
+      const item = window.localStorage.getItem(key)
+      setStoredValue(item ? JSON.parse(item) : initialValue)
+    } catch (error) {
+      console.log(error)
+      return setStoredValue(initialValue)
+    }
+  }, [])
+
+  return [storedValue, setValue]
```

Comment:
@pettinarip why is that needed?

Reply:
That tells TS that you are returning a fixed array (tuple). In this case, it will return an array, where the 1st position is of type `T` and the 2nd is a `Function`.

Reply:
Thx, fixed